### PR TITLE
[Merged by Bors] - feat: port Algebra.BigOperators.Intervals

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Intervals.lean [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+  rw [Nat.Ico_succ_right_eq_insert_Ico hab, sum_insert right_not_mem_Ico, add_comm]
+#align finset.sum_Ico_succ_top Finset.sum_Ico_succ_top
+
+@[to_additive]
+theorem prod_Ico_succ_top {a b : ℕ} (hab : a ≤ b) (f : ℕ → β) :
+    (∏ k in Ico a (b + 1), f k) = (∏ k in Ico a b, f k) * f b :=
+  @sum_Ico_succ_top (Additive β) _ _ _ hab _
+#align finset.prod_Ico_succ_top Finset.prod_Ico_succ_top
+-- porting note: this seems to be defined already above
+--#align finset.sum_Ico_succ_top Finset.sum_Ico_succ_top
```
Comment (jcommelin) [inline Mathlib/Algebra/BigOperators/Intervals.lean:63]: Yeah, that's a glitch in mathport.
```suggestion
```

File: Mathlib/Algebra/BigOperators/Intervals.lean [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+  rw [← sum_insert ha, Nat.Ico_insert_succ_left hab]
+#align finset.sum_eq_sum_Ico_succ_bot Finset.sum_eq_sum_Ico_succ_bot
+
+@[to_additive]
+theorem prod_eq_prod_Ico_succ_bot {a b : ℕ} (hab : a < b) (f : ℕ → β) :
+    (∏ k in Ico a b, f k) = f a * ∏ k in Ico (a + 1) b, f k :=
+  @sum_eq_sum_Ico_succ_bot (Additive β) _ _ _ hab _
+#align finset.prod_eq_prod_Ico_succ_bot Finset.prod_eq_prod_Ico_succ_bot
+-- porting note: this seems to be defined already above
+---#align finset.sum_eq_sum_Ico_succ_bot Finset.sum_eq_sum_ico_succ_bot
```
Comment (jcommelin) [inline Mathlib/Algebra/BigOperators/Intervals.lean:77]:
```suggestion
```

File: Mathlib/Algebra/BigOperators/Intervals.lean [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+/-!
+# Results about big operators over intervals
+
+We prove results about big operators over intervals (mostly the `ℕ`-valued `Ico m n`).
+-/
+
+
+universe u v w
+
+-- porting note: removed `BigOperators`
```
Comment (jcommelin) [inline Mathlib/Algebra/BigOperators/Intervals.lean:25]:
```suggestion
-- open BigOperators -- porting note: notation is global for now
```
(This way we can still find the string if we search for `open BigOperators`)

**Comment by jcommelin** [comment]
bors d+

**Comment by mo271** [comment]
thanks
bors r+

**Pull request closed without merging**